src/scripts/assume-role-with-web-identity.sh

```diff
@@ -1,5 +1,22 @@
 PARAM_ROLE_SESSION_NAME=$(eval echo "${PARAM_ROLE_SESSION_NAME}")
 
+if [ -z "${PARAM_ROLE_SESSION_NAME}" ]; then
+    echo "Role session name is required"
+    exit 1
+fi
+
+# shellcheck disable=SC2034
+AWS_STS_COMMAND=$(
```

Comment:
this looks to me like it'll attempt to assume role twice, both here on line 9 and again on line 21?   i'll admit to not being familiar with the pipe status exiting you're doing.  curious on how this works :)

Reply:
I took the dev published version of this orb via this PR and tested on our side.  While it works, I confirm our CloudTrail logs show duplicate assume roles with web identities with this code?

<img width="661" alt="image" src="https://user-images.githubusercontent.com/33203301/166495684-95d62160-0b17-45ea-8053-4f47b5c533fa.png">

This of course is not ideal as it means double audit logs; every user will look like they're logging in twice with OpenID.  So this needs to somehow be refactored into a single call I believe?

Reply:
(and btw i did note in my issue report that the root here, the failing when there is a failed Assume Role for Web Identity, now correctly fails; so we just want this to not double authenticate)

Reply:
@adamdmharvey, thanks for letting me know, I really appreciate it! Having double logs is definitely not ideal. Let me investigate some more and see what we can do.

Reply:
@adamdmharvey - I don't know why I had to make is so hard :joy: I've added validation that checks to see if the keys and session token were actually generated. If the `aws sts assume-role-with-web-identity` command fails, the keys and token will be null. I tested it by forcing it to fail and it gave me the correct error from `AWS` and the job did fail.

![image](https://user-images.githubusercontent.com/64455338/166630721-b80bd112-f425-4ba3-b8aa-385f52e54d76.png)

As for the executor, we decided to leave it in case there are others using it.